pkgs/by-name/inventaire-unwrapped/package.nix

```diff
@@ -0,0 +1,160 @@
+      --replace-fail \
+        "import type { SearchRequest } from '@elastic/elasticsearch/lib/api/types.js'" \
+        "import type { estypes } from '@elastic/elasticsearch'" \
+      --replace-fail 'SearchRequest' 'estypes.SearchRequest' \
+
+    substituteInPlace server/lib/search_by_position.ts \
+      --replace-fail \
+        "import type { SearchRequest } from '@elastic/elasticsearch/lib/api/types.js'" \
+        "import type { estypes } from '@elastic/elasticsearch'" \
+      --replace-fail 'SearchRequest' 'estypes.SearchRequest' \
+
+    # Look up some directories relative to CWD, instead of in the installed tree
+
+    substituteInPlace server/lib/auto_rotated_keys.ts \
+      --replace-fail "absolutePath('root', 'keys/sessions_keys')" "'keys/sessions_keys'"
+
+    substituteInPlace server/controllers/images/lib/local_client.ts \
+      --replace-fail 'resolve(projectRoot, localStorage.directory)' 'localStorage.directory'
+
+    # Please, don't run git to try to find out the revision of the src
+    substituteInPlace server/lib/package.ts \
+      --replace-fail "await execAsync('git rev-parse --short HEAD').then(({ stdout }) => stdout.trim())" "'${
+        if src.tag != null then src.tag else src.rev
+      }'" \
+
+    # OpenSearch is not happy with these API URLs, ElasticSearch seems to be fine with the patched ones as well
+    substituteInPlace server/db/elasticsearch/init.ts \
+      --replace-fail '/_doc/wikidata/test' '/wikidata/_doc/test'
+    substituteInPlace server/db/elasticsearch/bulk.ts \
+      --replace-fail 'Origin}/_doc/_bulk' 'Origin}/_bulk'
+  '';
+
+  makeCacheWritable = true;
+
+  nativeBuildInputs = [
+    tsx
+  ];
+
+  postInstall = ''
+    cp -r dist $out/lib/node_modules/inventaire/
+
+    # One link wants to point at inventaire-client, most of the others at generated files
+    # Just delete broken ones for now, and create empty dirs in their place
+    for candidate in $out/lib/node_modules/inventaire/dist/*; do
+      if [ -L "$candidate" ]; then
+        linkName="$(basename "$candidate")"
+        rm -v "$candidate"
+        if [ -e "$(dirname "$candidate")/../''${linkName}" ]; then
+          ln -vs ../"$linkName" "$candidate"
+        fi
+      fi
+    done
+
+    # Launcher
+    mkdir -p $out/bin
+    cat <<EOF >$out/bin/inventaire
+    #!/bin/sh
+
+    # Find server.js relative to location of this file
+    INVENTAIRE_THIS="\$0"
+    INVENTAIRE_SERVER="\$(dirname "\$INVENTAIRE_THIS")/../lib/node_modules/inventaire/dist/server/server.js"
+
+    if [ ! -e "\$INVENTAIRE_SERVER" ]; then
+      echo "Missing: \$INVENTAIRE_SERVER"
+
+      # Fallback to hardcoding (this will likely fail to run due to missing client symlink from the wrapper)
+      INVENTAIRE_SERVER="$out/lib/node_modules/inventaire/dist/server/server.js"
+      echo "Falling back to: \$INVENTAIRE_SERVER"
+    fi
+
+    exec ${lib.getExe nodejs} "\$INVENTAIRE_SERVER"
+    EOF
+    chmod +x $out/bin/inventaire
```

Comment:
This is redundant since mode is not preserved in `pkgs/by-name/inventaire/package.nix`, right? This said, why don't we create `$out/bin/inventaire` in there instead (the wrapped inventaire)?

Reply:
Can do.